### ISSUE TITLE
U-Net data types

### ISSUE THREAD

**Issue by Jhsmit** [HenriquesLab/ZeroCostDL4Mic#3]
Hi, I was wondering what the best way is of getting images with the appropriate data types in to the U-Net. 
I have a set of grayscale images (uint32) and binary images as labels. 
I think I've now gotten them into the UNet properly by saving as uint16 and uint8, respectively, using tifffile with the kwarg ``imagej=True``

I suspect my binary images get converted to all zeroes because in the training of the network the accuracy rapidly becomes 1 and the loss 0. The output I get is also all zero.

What is the correct datatype / format for the input images? I've added a zip with some of my input images.

[input.zip](https://github.com/HenriquesLab/ZeroCostDL4Mic/files/4377081/input.zip)

**Comment by Romain-Laine** [comment]
Hi,
It's a good point. If you look at the training dataset that we tested it on (from ISBI), the images are 8-bit TIF both for the input images and the masks. Additionally, the binary masks are set to 0 and 255 for black and white respectively. See below. So try and match this formatting and see whether it works for you!

Romain

<img width="1054" alt="Screen Shot 2020-03-25 at 10 25 40" src="https://user-images.githubusercontent.com/21193399/77526823-17f62580-6e83-11ea-9ab9-4c469d7ba876.png">

**Comment by Jhsmit** [comment]
I've changed my input binary to have values of either 0 or 255 and I've renormalized my brightfield images between 0 and 255 and changed the dtype to uint8. 
This seems to help, I do get some kind of output now that resembles what i'm after. The val_loss reached a plateau after a few epoch so I stopped training. 


![image](https://user-images.githubusercontent.com/7881506/77536229-521ff100-6e9c-11ea-86b7-d851b82c8e71.png)

The input images are very noisy but I know its possible to segment them with UNet. So probably some hyperparameter optimization required.

Cheers,
Jochem

PS. Addition of normalization options might be a nice feature
PPS. I've just found that you do some normalization:

``` python
def adjustData(img,mask,flag_multi_class,num_class):
    img = img / 255
    mask = mask /255
    mask[mask > 0.5] = 1
    mask[mask <= 0.5] = 0
    return (img,mask)
```

This function will make binary input images with 0/1 values all equal to 0.
